core/components/atoms/tooltip/tooltip.md

```diff
@@ -44,3 +44,54 @@ The Tooltip wraps long text by default to make it readable. Try to keep the cont
   <Icon name="help" />
 </Tooltip>
 ```
+
+### Tooltip for input actions
+
+You can use a special variation of the tooltip which is made for input actions.
+
+It does recieve a `content` object with the following keys:
+
+- `default`: Is the default text to show in the tooltip (same as `content` in a normal tooltip).
+- `loading`: Is the text will show if the child button `onClick` handler is an async function (Promise) while it's loading.
+- `success`: Is the text will show when the `onClick` returns (if it's a sync function) or resolves (if it's async).
+- `error`: If the text will show if the `onClick` is an async function and is rejected.
+
+#### Example
+
+```js
+<TextInput
+  type="text"
+  placeholder="Enter some text"
+  actions={[
+    <Tooltip.Action
+      content={{
+        default: 'Copy',
+        success: 'Copied!'
+      }}
+    >
+      <Button icon="copy" onClick={e => console.log(e)} />
+    </Tooltip.Action>,
+    <Tooltip.Action
+      content={{
+        default: 'Do work',
+        loading: 'Working...',
+        success: 'Done!',
+        error: 'Could not work'
+      }}
+    >
+      <Button icon="wrench" onClick={sampleAsyncAction} />
+    </Tooltip.Action>,
+    <Tooltip.Action
+      content={{
+        default: 'Do work',
+        loading: 'Working...',
+        success: 'Done!',
+        error: 'Could not work'
+      }}
+    >
+      <Button icon="reload" onClick={sampleAsyncFailingAction} />
```

Comment:
could we have an example where the button has a disabled/loading state as well?

Reply:
Done ✅